### PR TITLE
Add dpos module initials - Closes #6718

### LINKED ISSUE #6718: Create class, constants and schemas for DPoS module
### Description

Create DPoS module class, schemas and constants

- Define constants at `constants.ts`
- Define a DPoS module class with necessary initial variables that base module requires in `module.ts`
- Export module in `index.ts`
- Create an empty endpoint class in `endpoint.ts`
- Define the interface for the dependency modules and add dependency function in the class (Validators module, random module and BFT module)
- Define schemas in `schemas.ts`
- Define config schema in `schemas.ts` and bind it to module class
- Define necessary interfaces(e.g. dependency interfaces) in `types.ts`
- Define empty API class and bind it to module class with module id

### Acceptance Criteria

- All files should exist

### PR DESCRIPTION
**Pull request by mehmetegemen**
### What was the problem?

This PR resolves #6718 

### How was it solved?

- Added initial files

### How was it tested?

- Checked files manually

### CODE REVIEW

**Review by shuse2 — COMMENTED** [review]
LGTM except the naming.
- `DPOS` => `DPoS` 
- for the schema names, `XXXParams` => `XXXParamsSchema` and $id should include `Params` at the end

File: framework/src/modules/dpos_v2/commands/pom.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
```
Comment (ishantiw) [inline framework/src/modules/dpos_v2/commands/pom.ts:14]:
```suggestion

```

File: framework/src/modules/dpos_v2/commands/unlock.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
```
Comment (ishantiw) [inline framework/src/modules/dpos_v2/commands/unlock.ts:13]:
```suggestion

```

**Review by ishantiw — CHANGES_REQUESTED** [review]

File: framework/src/modules/dpos_v2/commands/unlock.ts (hunk 2 above)
Comment (mehmetegemen) [reply framework/src/modules/dpos_v2/commands/unlock.ts:13]: done

File: framework/src/modules/dpos_v2/commands/pom.ts (hunk 1 above)
Comment (mehmetegemen) [reply framework/src/modules/dpos_v2/commands/pom.ts:14]: done

**Review by shuse2 — APPROVED** [review]

**Review by ishantiw — APPROVED** [review]
🚀

**Pull request merged**